### PR TITLE
Add metricsmd package

### PR DESCRIPTION
**Pull request by lambdanis**
`pkg/metricsmd` is a utility for generating a metrics reference in the markdown format from initialized Prometheus registry. It's intended to be used as an add-on to applications that have a Cobra CLI and expose Prometheus metrics.

For example usage see https://github.com/cilium/tetragon/pull/2164

### CODE REVIEW

**Review by mtardy — APPROVED** [review]
Thanks! I mostly checked the last commits with the options and it's a good addition :)

File: .gitattributes
Comment (mtardy) [inline .gitattributes:1]: I think GitHub automatically does this appropriately?

Comment (lambdanis) [reply .gitattributes:1]: oh, does it? That's cool.

Comment (mtardy) [reply .gitattributes:1]: ah `go.sum` is always displayed but vendored stuff is redacted except for appropriate stuff like `vendor/modules.txt`. See your PR for example https://github.com/cilium/tetragon/pull/2164/files.

**Pull request merged**